client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -186,8 +205,33 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        synchronizer.updateState((state, updates) -> {
+            if (state.getConfigState() != ReaderGroupState.ConfigState.READY) {
+                val oldConfig = state.getConfig();
+                val newConfig = state.getNewConfig();
+                manageSubscriptions(oldConfig, newConfig);
```

Comment:
if manageSubscriptions throws an error, have  you verified that the error is thrown here and update is duly failed?

Reply:
Could you elaborate?